### PR TITLE
Add createWebhookEndpoint

### PR DESCRIPTION
**Pull request by supercrafter100**
This adds the ability to create a webhook endpoint via the api. Useful for third party applications that want to integrate with the api and create webhooks that send info back to their own webserver.

### CODE REVIEW

File: modules/Core/includes/endpoints/CreateWebhooksEndpoint.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                Validate::REQUIRED => true,
+            ]
+        ])->messages([
+            'name' => CoreApiErrors::ERROR_WEBHOOK_NAME_INCORRECT_LENGTH,
+            'url' => CoreApiErrors::ERROR_WEBHOOK_URL_INCORRECT_LENGTH
+        ]);
+
+        // If it didn't pass, throw the errors
+        if (!$validation->passed())
+            $api->throwError($validation->errors()[0]);
```
Comment (tadhgboyle) [inline modules/Core/includes/endpoints/CreateWebhooksEndpoint.php]:
```suggestion
        if (!$validation->passed()) {
            $api->throwError($validation->errors()[0]);
        }
```

File: modules/Core/includes/endpoints/CreateWebhooksEndpoint.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        DB::getInstance()->insert('hooks', [
+            'name' => $name,
+            'action' => $type,
+            'url' => $url,
+            'events' => json_encode($events)
+        ]);
+
+        // Clear cache so the webhooks are refreshed
+
+        // Annoying that there isn't a global way to get the nameless cache
```
Comment (tadhgboyle) [inline modules/Core/includes/endpoints/CreateWebhooksEndpoint.php]:
```suggestion
```

File: modules/Core/includes/endpoints/CreateWebhooksEndpoint.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    public function __construct() {
+        $this->_route = 'webhooks/create';
+        $this->_module = 'Core';
+        $this->_description = 'Create a new webhook';
+        $this->_method = 'POST';
+    }
+
+    public function execute(Nameless2API  $api): void {
+        // Validation
+        $api->validateParams($_POST, ['name', 'url', 'type', 'events']);
```
Comment (tadhgboyle) [inline modules/Core/includes/endpoints/CreateWebhooksEndpoint.php]: Does the `Validate::check(...)` following this not do the same thing?

Comment (supercrafter100) [reply modules/Core/includes/endpoints/CreateWebhooksEndpoint.php]: It does I believe. Will remove that

File: modules/Core/includes/endpoints/CreateWebhooksEndpoint.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    public function __construct() {
+        $this->_route = 'webhooks/create';
+        $this->_module = 'Core';
+        $this->_description = 'Create a new webhook';
+        $this->_method = 'POST';
+    }
+
+    public function execute(Nameless2API  $api): void {
+        // Validation
+        $validation = Validate::check($_POST, [
```
Comment (partydragen) [inline modules/Core/includes/endpoints/CreateWebhooksEndpoint.php:22]: Validation missing "events" perhaps also make Validation check IS ARRAY?

Atleast these errors are currently not used with they should be
public const ERROR_WEBHOOK_INVALID_TYPE = 'core:webhook_type_invalid';
public const ERROR_WEBHOOK_INVALID_EVENT = 'core:webhook_event_invalid';

Comment (supercrafter100) [reply modules/Core/includes/endpoints/CreateWebhooksEndpoint.php:22]: Added validation for these terms

**Review by partydragen — APPROVED** [review]

**Review by samerton — APPROVED** [review]

**Pull request merged**